schemas/iso19139/src/main/java/org/fao/geonet/schema/iso19139/FormatReplacer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2001-2016 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+package org.fao.geonet.schema.iso19139;
+
+import java.util.List;
+
+import org.fao.geonet.kernel.schema.subtemplate.AbstractReplacer;
+import org.fao.geonet.kernel.schema.subtemplate.ConstantsProxy;
+import org.fao.geonet.kernel.schema.subtemplate.ManagersProxy;
+import org.fao.geonet.kernel.schema.subtemplate.SubtemplatesByLocalXLinksReplacer;
+import org.jdom.Element;
+import org.jdom.JDOMException;
+import org.jdom.Namespace;
+
+public class FormatReplacer extends AbstractReplacer {
+
+    public FormatReplacer(List<Namespace> namespaces,
+                          ManagersProxy managersProxy,
+                          ConstantsProxy constantsProxy) {
+        super(namespaces, managersProxy, constantsProxy);
+    }
+
+    @Override
+    public String getAlias() {
+        return SubtemplatesByLocalXLinksReplacer.FORMAT;
+    }
+
+    @Override
+    protected String getElemXPath() {
+        return ".//gmd:MD_Format/parent::*";
+    }
+
+    @Override
+    protected int queryAddExtraClauses(StringBuilder query, Element format, String lang) throws JDOMException {
+        String name = getFieldValue(format, ".//gmd:name", lang);
```

Comment:
Instead of hardcoding the replacer config, should we do something more like 

```xml
  <bean id="iso19139SchemaPlugin"
        class="org.fao.geonet.schema.iso19139.ISO19139SchemaPlugin">
    <property name="replacers">
      <util:list>
        <bean id="iso191139FormatReplacer" class="Replacer">
          <property name="xpath" value=".//gmd:MD_Format/parent::*"/>
          <property name="matching">
            <util:map>
              <entry key="fname" value=".//gmd:name"/>
              <entry key="fversion" value=".//gmd:version"/>
            </util:map>
          </property>
<!--          <property name="extraClause" value="+valid:1"/> Allow to limit subtemplates to choose from -->
<!--          <property name="minWeight" value="?"/> Not really sure how this is suppose to work yet. If not set, then do exact match (search will BTW not always be exact - depends on analyzers) -->
        </bean>
        ...
      </util:list>
    </property>
```